### PR TITLE
Update viewer.php

### PR DESCRIPTION
**Pull request by horosgrisa**
Fix Close button

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@jancborchardt css-fu - THX

**Comment by jancborchardt** [comment]
@horosgrisa can you shortly describe which problem it fixes, ideally with a before-after screenshot? :) Or refer the issue it fixes? Thanks!

**Comment by jancborchardt** [comment]
@horosgrisa as far as I know the pdf viewer does not use the standard close icon. Would be good if you can switch that with the core one by using `class="icon-close"`. :)

You are also very welcome in our IRC channel at #owncloud-design!

**Comment by horosgrisa** [comment]
### Before

### After

**Comment by jancborchardt** [comment]
@horosgrisa aah, great stuff! Do you mind changing the icon to the proper icon-close one as well?

**Comment by horosgrisa** [comment]
Yes, I now want to do it. Now used letter "X". But, I don't have good icon for this...

**Comment by LukasReschke** [comment]
@jancborchardt The ownCloud CSS is not included in the PDF viewer since otherwise a lot of things would explode. Thus there is only this small X at the moment :see_no_evil:

**Comment by jancborchardt** [comment]
@horosgrisa you can use this close icon: https://github.com/owncloud/core/blob/master/core/img/actions/view-close.svg – it’s white, and maybe you need to force it to 16px. :)
PNG at https://github.com/owncloud/core/blob/master/core/img/actions/view-close.png – you can just copy both into the PDF viewer app.

**Comment by horosgrisa** [comment]
@jancborchardt Тhanks, now pdfviewer looks like that

### Before
before

### After
after

File: js/previewplugin.js [hunk 1]
```diff
@@ -96,9 +97,9 @@
 					if($('#isPublic').val()) {
 						var sharingToken = $('#sharingToken').val();
 						downloadUrl = OC.generateUrl('/s/{token}/download?files={files}&path={path}', {
-							token: sharingToken,
-							files: fileName,
-							path: context.dir
+							token: encodeURIComponent(sharingToken),
+							files: encodeURIComponent(fileName),
+							path:  encodeURIComponent(context.dir)
```
Comment (LukasReschke) [inline js/previewplugin.js]: Encoding changes look unrelated. Please double-check.

File: js/previewplugin.js [hunk 2]
```diff
@@ -118,7 +119,7 @@ OC.Plugins.register('OCA.Files.FileList', OCA.FilesPdfViewer.PreviewPlugin);
 $(document).ready(function(){
 	if ($('#isPublic').val() && $('#mimetype').val() === 'application/pdf') {
 		var sharingToken = $('#sharingToken').val();
-		var downloadUrl = OC.generateUrl('/s/{token}/download', {token: sharingToken});
+		var downloadUrl = OC.generateUrl('/s/{token}/download', {token: encodeURIComponent(sharingToken)});
```
Comment (LukasReschke) [inline js/previewplugin.js]: Encoding changes look unrelated. Please double-check.

File: js/previewplugin.js [hunk 3]
```diff
@@ -43,7 +43,8 @@
 		show: function(downloadUrl, isFileList) {
 			var self = this;
 			var $iframe;
-			var viewer = OC.generateUrl('/apps/files_pdfviewer/?file={file}', {file: downloadUrl});
+
+			var viewer = OC.generateUrl('/apps/files_pdfviewer/?file={file}', {file: encodeURIComponent(downloadUrl)});
```
Comment (LukasReschke) [inline js/previewplugin.js]: Encoding changes look unrelated. Please double-check.

File: vendor/pdfjs/web/viewer.css [hunk 4]
```diff
@@ -472,13 +472,13 @@ html[dir='rtl'] .doorHangerRight:before {
```
Comment (LukasReschke) [inline vendor/pdfjs/web/viewer.css]: Can you move this to a new css file and include this instead? – Otherwise this will get overwritten in the next sync and nobody will remember the manual changes :-)

For that just create a folder `css` in the apps root folder, create a file such as `style.css` and include it in the `viewer.php` – let us know if you need help :smile:

File: vendor/pdfjs/web/viewer.css [hunk 5]
```diff
@@ -857,6 +857,16 @@ html[dir='rtl'] .toolbarButton#secondaryToolbarToggle::before {
   content: url(images/toolbarButton-secondaryToolbarToggle-rtl.png);
 }
 
+/* FIXME  */
+html[dir='ltr'] .toolbarButton#secondaryToolbarClose::before {
+  content: url(images/toolbarButton-secondaryToolbarClose.png);
+}
+html[dir='rtl'] .toolbarButton#secondaryToolbarClose::before {
+  content: url(images/toolbarButton-secondaryToolbarClose-rtl.png);
+}
```
Comment (jancborchardt) [inline vendor/pdfjs/web/viewer.css]: You don’t need two separate icons here depending on ltr or rtl, no?

File: vendor/pdfjs/web/viewer.css [hunk 6]
```diff
@@ -1687,6 +1697,17 @@ html[dir='rtl'] #documentPropertiesOverlay .row > * {
     content: url(images/toolbarButton-secondaryToolbarToggle-rtl@2x.png);
   }
 
+/* FIXME */
+
+ html[dir='ltr'] .toolbarButton#secondaryToolbarClose::before {
+    content: url(images/toolbarButton-secondaryToolbarClose@2x.png);
```
Comment (jancborchardt) [inline vendor/pdfjs/web/viewer.css]: If you just use the original SVG I linked you, you do not even need the double resolution icon. SVG is scalable by default. Just give the element of the image the class "svg" and ownCloud will handle the PNG fallback. (If you have the close.svg and close.png in the same folder).

**Comment by jancborchardt** [comment]
Looks good design-wise! The code is a bit complicated though as I commented – you can just use the SVG instead of manually doing the 2x png. And rtl/ltr is not really a difference here.

**Comment by horosgrisa** [comment]
@jancborchardt I have a problem with Content Security Policy. Similar to https://github.com/owncloud/calendar/issues/599 in master branch

**Comment by jancborchardt** [comment]
@horosgrisa what do you exactly mean? Our security expert @LukasReschke might be able to help here.

**Comment by horosgrisa** [comment]
I'm using Firefox, and I have the following problems

```
An iframe which has both allow-scripts and allow-same-origin for its sandbox attribute can remove its sandboxing. files
The site specified an invalid Strict-Transport-Security header.[Learn More]
```

after2

**Comment by horosgrisa** [comment]
I think that problem in file `controller/displaycontroller.php`, in following lines
```php
use OCP\AppFramework\Http\ContentSecurityPolicy;
```
and
```php
$response = new TemplateResponse($this->appName, 'viewer', $params, 'blank');

$policy = new ContentSecurityPolicy();
$policy->addAllowedChildSrcDomain('\'self\'');
$policy->addAllowedFontDomain('data:');
$response->setContentSecurityPolicy($policy);

return $response;
```
In `stable8` branch we have only 
```php
return new TemplateResponse($this->appName, 'viewer', $params, 'blank');
```
and this fork fine

**Comment by LukasReschke** [comment]
> An iframe which has both allow-scripts and allow-same-origin for its sandbox attribute can remove its sandboxing. files
> The site specified an invalid Strict-Transport-Security header.[Learn More]

Which Firefox version are you using? – Works fine here. Can you also provide me with a test account to your instance? (lukas at owncloud dot com)

**Comment by horosgrisa** [comment]
@LukasReschke I create an account for you at my test server with latest version of `files_pdfviewer` from `master` branch, and I send account credentials to your e-mail.

**Comment by LukasReschke** [comment]
> @LukasReschke I create an account for you at my test server with latest version of files_pdfviewer from master branch, and I send account credentials to your e-mail.

That makes sense, the master branch of this code only works with ownCloud master (aka upcoming 8.1), in this case the code is trying to access new public API introduced with 8.1 and fails hard.

The easiest way to check this is to take a look at the network console, a 500 error usually indicates a PHP error:


And looking at the ownCloud indeed confirms this:



You do have multiple options:

1. Setup an ownCloud development environment on master. This can be done by running `git clone --recursive https://github.com/owncloud/core.git` to a path accessible to a web server and then cd into it's app directory and run `git clone https://github.com/owncloud/files_pdfviewer.git` and enable the PDF viewer. This will leave you with the newest code. You can also pull down your own fork and sync it from time to time with upstream: https://help.github.com/articles/syncing-a-fork/
2. Reverting https://github.com/owncloud/files_pdfviewer/commit/afd3cab473dbf1a1def17b512870d52e3ca56e27 locally for you but submit the changes upstream again without these changes

Please let us know if you need any help or so please ask, we always are more than happy to support and help new contributors - you are more than welcome in #owncloud-dev on Freenode IRC :-)

**Comment by horosgrisa** [comment]
@LukasReschke Yes, it works fine at ownCloud 8.1. You can check this now using same credentials.

**Comment by LukasReschke** [comment]
Works, tested in single sharing view, folder sharing view and normal ownCloud view :+1:

Thanks a lot for your contribution @horosgrisa!

**Comment by LukasReschke** [comment]
@jancborchardt Second review?

**Comment by jancborchardt** [comment]
@horosgrisa instead of using the @2x resolution png, best just use the SVG.

**Comment by jancborchardt** [comment]
And the rtl / ltr CSS code can be safely removed, no?

**Comment by horosgrisa** [comment]
@jancborchardt I deleted the rtl CSS code and png images, now used svg

**Comment by jancborchardt** [comment]
@owncloud/designers @tomneedham can you double-check please?

**Comment by horosgrisa** [comment]
@jancborchardt, Hi, I remove unnecessary 'Open file' button

File: css/style.css [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  height: 16px;
+  width:16px;
+
+}
+
+
+html[dir='ltr'] .doorHangerRight:after {
+  right: 53px!important;
+}
+html[dir='ltr'] .doorHangerRight:before {
```
Comment (jancborchardt) [inline css/style.css]: This ltr code here is still not needed.

**Comment by LukasReschke** [comment]
Anyone? Moving to 8.2 for now…

**Comment by jancborchardt** [comment]
@owncloud/designers @tomneedham please double-check, thanks! :)

**Comment by DeepDiver1975** [comment]
:+1:
